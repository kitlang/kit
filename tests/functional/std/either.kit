function main() {
    var a: Either[Int, CString] = Left(123);

    match a {
        Left(x) => {
            printf("expected: %i\n", x);
        }
        default => {
            printf("unexpected!\n");
        }
    }

    a = Right("stringvalue");
    match a {
        Right(x) => {
            printf("expected: %s\n", x);
        }
        default => {
            printf("unexpected!\n");
        }
    }
}
