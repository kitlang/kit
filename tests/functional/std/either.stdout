expected: 123
expected: stringvalue
